Codigo/MODULO_NPCs.bas

```diff
@@ -43,16 +43,53 @@ ErrHandler_InitizlizeNpcIndex:
     Call TraceError(Err.Number, Err.Description, "NPCs.InitializeNpcIndexHeap", Erl)
 End Sub
 
-Public Function ReleaseNpc(ByVal NpcIndex As Integer) As Boolean
+Public Function IsValidNpcRef(ByRef NpcRef As t_NpcReference) As Boolean
+    IsValidNpcRef = False
+    If NpcRef.ArrayIndex <= 0 Or NpcRef.ArrayIndex > UBound(NpcList) Then
```

Comment:
Instead of `If NpcRef.ArrayIndex <= 0` maybe better `If NpcRef.ArrayIndex < LBound(NpcList)`